### ISSUE TITLE
Running 'extract' with method 'MIREOT' with erroneous syntax after --upper-term gives IndexOutOfBoundsException

### ISSUE THREAD

**Issue by agbeltran** [ontodev/robot#247]
The following command is syntactically wrong

```
robot extract --method MIREOT -I "http://purl.obolibrary.org/obo/duo.owl" --upper-term "obo:DUO_0000001" "obo:DUO_0000017" --lower-term "obo:DUO_0000001" "obo:DUO_0000017" --output output.owl
```
as there are two upper terms declared without repeating the option ```--upper-term```.

However, instead of getting a command error, I am getting an ```IndexOutOfBoundsException```:

```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at org.obolibrary.robot.MergeOperation.merge(MergeOperation.java:53)
	at org.obolibrary.robot.MergeOperation.merge(MergeOperation.java:65)
	at org.obolibrary.robot.ExtractCommand.execute(ExtractCommand.java:153)
	at org.obolibrary.robot.CommandManager.executeCommand(CommandManager.java:219)
	at org.obolibrary.robot.CommandManager.execute(CommandManager.java:161)
	at org.obolibrary.robot.CommandManager.main(CommandManager.java:121)
	at org.obolibrary.robot.CommandLineInterface.main(CommandLineInterface.java:53)
```

This is due to ```ExtractCommand``` calling the ```MergeOperation``` with an empty ```ArrayList```, see:
https://github.com/ontodev/robot/blob/master/robot-command/src/main/java/org/obolibrary/robot/ExtractCommand.java#L153

**Comment by jamesaoverton** [comment]
You're right, this is very bad behaviour.

@rctauber is working on a PR for better error reporting right now. I'll ask her to fix this too.

**Comment by agbeltran** [comment]
great, thanks!

**Comment by rctauber** [comment]
Our option parser was cutting off after the first `--upper-term` arg, and since the second arg was quoted, it wasn't identifying it as another command (in which case, it would have returned `unknown command or option: ...`).

I put a check in for after the first set of options are removed (remaining commands/options for chaining). If the first element of the `arguments` list is not a valid command name (as long as `arguments` is not empty), ROBOT will return the same message: `unknown command or option: ...`.

Here's what happens when I run your command now:
```
UNKNOWN INPUT ERROR unknown command or option: obo:DUO_0000017
See http://robot.obolibrary.org/errors#unknown-input-error for more details
{general usage information}
```

The link will provide more details on the assumption that options normally only accept one argument, with a few exceptions. It's a little more difficult to provide more details than that because of how chaining is handled, but hopefully this is enough for users to find the resources that they need.

I have some other cleaning up of the error reporting to do, but I'll push these changes to my [exception-handler](https://github.com/rctauber/robot/tree/exception-handler) branch later today.

**Comment by jamesaoverton** [comment]
Should be fixed by #246
